FAQ usability rewrite. (#116)
OpenTabletDriver/OpenTabletDriver.Web: OpenTabletDriver.Web/Views/Wiki/FAQ/Linux.md

Please reboot before attempting the solutions below to avoid misconfiguration.

#### It is a supported tablet, but still doesn't detect after a reboot?

Look for a log that resembles the following:

<small class="text-muted">Hint: You can see the console log under the "Console" tab in OpenTabletDriver.UX. You will better find problems when you
Switch from `Info` to `Debug` mode in the bottom left of the Console panel.
</small>

##### hid_uclogic kernel module interference {#hid_uclogic}

> Another tablet driver found: UC Logic\
> ArgumentOutOfRangeException Value range is [0, 15]. (Parameter 'value')

These errors may happen if the `hid_uclogic` kernel module is loaded at the same time as OpenTabletDriver.

You can remove and blacklist this module to stop it from loading in the future by running the following commands:
```bash
echo "blacklist hid_uclogic" | sudo tee -a /etc/modprobe.d/blacklist.conf
sudo rmmod hid_uclogic
```

##### Missing udev rules {#udev}

> Not permitted to open HID class device at /dev/hidrawX

If you have followed the previous instructions, usually this is the only reason OpenTabletDriver will fail to detect your tablet when it's supported.

> The issue is resolved by making sure that you have proper udev rules set up.

Reviewer: ```suggestion
This issue can be resolved by ensuring that you have proper udev rules configured.
```

###### From Package Manager

If you installed OpenTabletDriver via your package manager, you may just need to reload and trigger the rules:
```bash
sudo udevadm control --reload-rules
sudo udevadm trigger
```

If it still doesn't work after this point, and you installed OpenTabletDriver with a package manager, please reboot, some systems may require this.

###### Portable install

If you installed OpenTabletDriver without using a package manager you will not have proper udev rules set up and must build your own.
It is highly recommended that you use your package manager where available.

> This process will require you to have the dotnet 7 SDK installed before proceeding:

Reviewer: I would hyperlink here to where the dotnet 7 sdk can be downloaded

Reviewer 2: This is distribution dependent, is it not? But if a link is needed, I suggest https://github.com/dotnet/core/blob/main/linux.md or better.

The "Install .NET on Linux instuctions" provided on the download page aren't great IMO, but it does link to the above page which I think is more safe, since it seems to prefer repo packages.

Author: If anything we should refer the user to their package manager, on Linux you should never really manage this yourself.

```bash
# Clone the repository, change current directory to the repository
git clone https://github.com/OpenTabletDriver/OpenTabletDriver.git
cd ./OpenTabletDriver

# Generate rules, moves them to the udev rules directory
./generate-rules.sh
sudo mv ./bin/99-opentabletdriver.rules /etc/udev/rules.d/99-opentabletdriver.rules

# Reload udev rules
sudo udevadm control --reload-rules

# Clean up leftovers
cd ..
rm -rf OpenTabletDriver
```
Then restart OpenTabletDriver and replug your tablet.

> #### It is not a supported tablet, what can I do about that?

Reviewer: ```suggestion
#### My tablet is not supported, what can I do about that?
```

If you want to add support for your tablet on your own, feel free to look at a similar configuration on our GitHub and our configuration documentation.

If you would like help with supporting your tablet, or would prefer us to do it, join our [Discord](/Discord) and create a post in `#config-creation` or a support channel.

### My cursor is teleporting? {#ghost-cursor}

This occurs when another program is reading the tablet at the same time as OpenTabletDriver, if this is happening everywhere on the system,
the solution to this is very simple:
```bash
echo "blacklist wacom" | sudo tee -a /etc/modprobe.d/blacklist.conf
sudo rmmod wacom
```

> If you are using a Wayland compositor, it may be required for you use artist mode to pass the tablet as a virtual tablet
> rather than an absolute mouse because certain applications like SDL and XWayland may incorrectly handle the cursor otherwise.
> This is a rather simple setup which you can find <a href="#artist-mode">here</a>.

Reviewer: ```suggestion
If you are using a Wayland compositor, it may be required for you to use artist mode to pass the tablet to the driver as a virtual tablet, rather than an absolute mouse - certain applications like SDL and XWayland may incorrectly handle the cursor otherwise.
This is a rather simple config change, which can be found <a href="#artist-mode">here</a>.
```

### I blacklisted kernel modules, but they load on startup? {#blacklisted-modules-loading}

This issue can be caused by either your initramfs not containing the instructions to not load the modules
or another crucial service loading the module despite it being blacklisted. You will have to find that service
yourself if the solution below does not solve your issue.

> Rebuilding the initramfs is an easy process, but the commands to do this vary between the distro.

Reviewer: ```suggestion
Rebuilding the initramfs is an easy process, but the commands to do this can vary between distributions of linux.
```

#### Arch

```bash
sudo mkinitcpio -P
```

#### Debian / Ubuntu
```bash
sudo update-initramfs -u
```

### My tablet detects, but it isn't working? {#fail-virtual-device}

If this is on a fresh install of OpenTabletDriver, usually problems like this can be solved
simply by restarting your computer. If it isn't resolved you may try resetting your OpenTabletDriver
settings to the defaults with `File > Reset to defaults`, then pressing save.

If your issue is still not resolved, and you are using OpenTabletDriver.UX, navigate to the "Console" tab
at the top of the main interface, switch from `Info` to `Debug` in the bottom left of that window and you will be see every log message.

> If you are using the daemon directly you may simply just look for a log

Reviewer: ```suggestion
If you are using the daemon directly, simply look for a log entry
```

your kernel config.

##### Gentoo
Check the option `CONFIG_INPUT_UINPUT` in /usr/src/linux.config. If it is not enabled, use the following steps

```bash
cd /usr/src/linux
sudo make menuconfig
```

Having done that, select y (for always active) or m (for building as a module) for the option in
*Device Drivers ---> Input Device Support ---> Miscellaneous devices ---> User level driver support* and rebuild the kernel.

##### Other Distros

Refer to your distribution's documentation regarding kernel configuration to enable this option.

This is probably due to your kernel not having uinput built in, if `stat /dev/uinput` doesn't return a file and you are using a custom kernel,
compile your kernel with support for uinput or use a different kernel.

### Poor performance with NVIDIA {#performance-nvidia}

Disable "Force full composition pipeline" in the NVIDIA settings panel.

### Stuck cursor in osu!lazer (Wayland) {#osu-lazer-broken-input-wayland}

Make sure you set the `SDL_VIDEODRIVER` to `wayland`:

```bash
env SDL_VIDEODRIVER=wayland ./osu.AppImage
```
or
```bash
env SDL_VIDEODRIVER=wayland osu-lazer
```

> 

Reviewer: ```suggestion
You can use either of these commands when creating a shortcut, script, or `.desktop` file to run osu! with, or add it to the existing runner you're already using to launch osu!.
```

Reviewer 2: Suggesting more than provided seems out of scope, since there is a chance users might misunderstand that they can freely edit system files (which the osu-lazer desktop file can be), undoing their changes eventually on a system update.

Author: I agree. We shouldn't suggest modifying system files.

### My CTL-x100 is not detected? {#CTL-x100-android-mode}

> It is possible for CTL-x100 tablets boots in Android mode instead of PC mode in some rare cases. To fix this, press

Reviewer: ```suggestion
It is possible for CTL-x100 tablets to boot in Android mode (the mode they use when interfacing with android devices like phones) instead of PC mode in some rare cases. To fix this, press
```

Instead, if you have installed a packaged build you can run `otd-daemon` in a terminal

If you are not using a packaged build and instead compiled from source,
you will need to run `./OpenTabletDriver.Daemon` in your terminal from the build directory instead.

You may place this in your desktop environment startup for it to automatically start.

### Enabling pressure using Artist Mode {#artist-mode}

Pressure support is available by changing the output mode of OpenTabletDriver to Artist Mode:

- Change output mode (at the bottom left of OpenTabletDriver) to Artist Mode.
- Remove the Tip Binding in the Pen Settings panel by clicking the 3 dots, then pressing clear.
- Save your settings and try drawing in an applicantion that supports pressure.

You will also need to use artist mode mouse bindings in the advanced binding editor (press '...' next to the binding)
instead of regular mouse buttons. Regular mouse buttons are not currently supported in artist mode.

### Disabling libinput's tablet device smooething

Using artist mode will result in some minor smoothing due to libinput's tablet handling.

> To disable this smoothing add the contents below to `/etc/libinput/local-overrides.quirks`:

Reviewer: ```suggestion
To disable this smoothing, add the contents below to `/etc/libinput/local-overrides.quirks`:
```
